ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterFactory.java

```diff
@@ -86,11 +91,18 @@ public NonBlockingRouterFactory(VerifiableProperties verifiableProperties, Clust
     MetricRegistry registry = clusterMap.getMetricRegistry();
     routerMetrics = new NonBlockingRouterMetrics(clusterMap, routerConfig);
     networkConfig = new NetworkConfig(verifiableProperties);
+    http2ClientConfig = new Http2ClientConfig(verifiableProperties);
     networkMetrics = new NetworkMetrics(registry);
+    http2ClientMetrics = new Http2ClientMetrics(registry);
+
     time = SystemTime.getInstance();
-    networkClientFactory = new SocketNetworkClientFactory(networkMetrics, networkConfig, sslFactory,
-        routerConfig.routerScalingUnitMaxConnectionsPerPortPlainText,
-        routerConfig.routerScalingUnitMaxConnectionsPerPortSsl, routerConfig.routerConnectionCheckoutTimeoutMs, time);
+    if (new ClusterMapConfig(verifiableProperties).clusterMapHttp2Enable) {
```

Comment:
After reviewing the whole PR, I actually wonder if we can put the `Http2ClientEnabled` config into `NetworkConfig`.   Any specific reason to keep it in the `ClusterMapConfig` ?

Reply:
I initially tried to add this in networkConfig, the problem is DataNode:getPortConnectTo(). It doesn't have NetworkConfig context.

Another benefit is we aimed to use this to override sslEnabledDc config in ClusterMapConfig.